slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbPruner.java

```diff
@@ -60,6 +60,10 @@ public void pruneForValidator(
   }
 
   private void pruneBlocks(final int validatorId, final long slotsToKeep) {
+
+    final Jdbi jdbi =
+        this.jdbi.orElseThrow(() -> new NullPointerException("Pruner has no connection created"));
```

Comment:
if (jdbi.isEmpty()) {
return; //no-op since jdbi is empty/pruning is disabled
}

Reply:
We've talked about and I'd rather throw an Exception here to let whoever is trying to call the pruner with no connection that something is wrong rather than just doing nothing.